subprojects/execution/src/main/java/org/gradle/internal/execution/model/annotations/InputPropertyAnnotationHandler.java

```diff
@@ -108,4 +111,18 @@ private void validateNotDirectoryType(PropertyMetadata propertyMetadata, TypeVal
             );
         }
     }
+    private static void validateNotUrlType(PropertyMetadata propertyMetadata, TypeValidationContext validationContext) {
+        List<Class<?>> valueTypes = unpackValueTypesOf(propertyMetadata);
+        if (valueTypes.stream().anyMatch(URL.class::isAssignableFrom)) {
+            validationContext.visitPropertyProblem(problem ->
+                    problem.withId(ValidationProblemId.UNSUPPORTED_VALUE_TYPE)
+                            .forProperty(propertyMetadata.getPropertyName())
+                            .reportAs(ERROR)
```

Comment:
:x: Going for `ERROR` directly here would mean a breaking change (your build works in 8.2 and it fails in 8.3), which we want to avoid. What we usually do is to introduce new validations as `WARNING`s.

They get reported as deprecation warnings here: https://github.com/gradle/gradle/blob/b1b2792f6f5251cddfaaf1efe87910b811f7166c/subprojects/core/src/main/java/org/gradle/execution/DefaultWorkValidationWarningRecorder.java#L49-L53

In the next major we turn them to `ERROR`s.

Reply:
Oops, that's right!

Reply:
This still needs to be changed, right?